### ISSUE TITLE
Add the ability to compare tags

### ISSUE THREAD

**Issue by barbuslex** [gogits/gogs#3621]
Hi,

Is it possible to implement in Gogs the ability to compare differences between two tags.

The git command is:
```
git log --pretty=format:%s oldTag..NewTag
```

It would become easy to generate a changelog for the new version of our project.

Thanks in advance.

++

**Comment by danowar2k** [comment]
I can't compare a tag with another branch. The above missing functionality is probably the reason for it...

**Comment by Unknwon** [comment]
@danowar2k no, because the current compare function treats all inputs are commit SHA, so tag name won't work.

**Comment by idan89** [comment]
Does anyone has a workaround to compare tags instead of commits? It this even possible?
Thanks.
